salt/pillar/__init__.py

```diff
@@ -692,7 +692,7 @@ def get_tops(self):
                 else:
                     saltenvs.add(self.opts["pillarenv"])
             else:
-                saltenvs = self._get_envs()
+                saltenvs = {*self._get_envs()}
```

Comment:
**question** would `saltenvs.update(self._get_envs())` be more accurate here?

Reply:
Hi @waynew : it's _probably_ more in keeping with the style of this function (given that the Set assigned to `saltenvs` on L681 is clearly destined for mutation). With that said, the line that I replaced was not attempting to mutate `saltenvs`, but rather, rebound the variable name to a new set (well, actually a list, due to the reported bug). This, combined with the fact that I usually attempt to avoid mutation led me to rebind it as well.

I'm happy to update the PR if your preference is to mutate - just let me know :)

Reply:
I'm +0 on it :joy: 

The Boy Scout in me says that we should go ahead and make the change, so that it isn't confusing anyone down the road.

The pragmatist in me says :shrug: 

I guess... if you're happy enough to make the change, then let's make the change and get this merged in :+1:

Reply:
FWIW, I went through the git history for this function and... it definitely had some changes. IIRC the *desired* behavior actually was to do some mutation

Reply:
No worries @waynew - I've just pushed that change now :)